src/Internal.AspNetCore.HostingStartup.Sdk/TrimDeps.cs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace RepoTasks
+{
+    public class TrimDeps : Task
+    {
+        [Required]
+        public ITaskItem[] DepsFiles { get; set; }
+
+        public override bool Execute()
+        {
+            foreach (var depsFile in DepsFiles)
+            {
+                ChangeEntryPointLibraryName(depsFile.GetMetadata("Identity"));
```

Comment:
FYI - this is identical to `depsFile.ItemSpec`. I have a slight preference for this instead of .GetMetadata, but its functionally the same.